GetVisibleServices/handler.ts

```diff
@@ -34,6 +35,76 @@ type IGetVisibleServicesHandlerRet =
 
 type IGetVisibleServicesHandler = () => Promise<IGetVisibleServicesHandlerRet>;
 
+const NationalServiceTuple = t.intersection([
+  ServiceTuple,
+  t.interface({
+    scope: t.literal(ServiceScopeEnum.NATIONAL)
+  })
+]);
+type NationalServiceTuple = t.TypeOf<typeof NationalServiceTuple>;
+
+const LocalServiceTuple = t.intersection([
+  ServiceTuple,
+  t.interface({
+    scope: t.literal(ServiceScopeEnum.LOCAL)
+  })
+]);
+type LocalServiceTuple = t.TypeOf<typeof LocalServiceTuple>;
+
+const ScopeGroupedServices = t.interface({
+  [ServiceScopeEnum.NATIONAL]: t.readonlyArray(NationalServiceTuple),
+  [ServiceScopeEnum.LOCAL]: t.readonlyArray(LocalServiceTuple)
+});
+type ScopeGroupedServices = t.TypeOf<typeof ScopeGroupedServices>;
+
+const VisibleServiceDictionary = t.dictionary(ServiceId, VisibleService);
+type VisibleServiceDictionary = t.TypeOf<typeof VisibleServiceDictionary>;
+
+/**
+ * Returns VisibleServices grouped by scope.
+ */
+const groupByScope = (
+  serviceJson: VisibleServiceDictionary
+): ScopeGroupedServices =>
+  toServicesTuple(new StrMap(serviceJson)).reduce(
+    (acc, service) => {
+      if (service.scope === ServiceScopeEnum.LOCAL) {
+        return {
+          [ServiceScopeEnum.NATIONAL]: acc[ServiceScopeEnum.NATIONAL],
+          [ServiceScopeEnum.LOCAL]: [
+            ...acc[ServiceScopeEnum.LOCAL],
+            (service as unknown) as LocalServiceTuple
+          ]
+        };
+      }
+      return {
+        [ServiceScopeEnum.NATIONAL]: [
+          ...acc[ServiceScopeEnum.NATIONAL],
+          (service as unknown) as NationalServiceTuple
+        ],
+        [ServiceScopeEnum.LOCAL]: acc[ServiceScopeEnum.LOCAL]
+      };
```

Comment:
```suggestion
      return {
        ...acc,
        [ServiceScopeEnum.NATIONAL]: [
          ...acc[ServiceScopeEnum.NATIONAL],
          (service as unknown) as NationalServiceTuple
        ]
      };
```

Reply:
Fixed in 6e9481e

Reply:
I actually realised you can maybe avoid all this by simply
```ts
return {
  ...acc,
  [service.scope]: [
    ...acc[service.scope],
    service
   ]
};
```
I haven't tried if the type is correctly inferred, though.

Reply:
It was, but IMHO is more readable using `ServiceScopeEnum.LOCAL`

Reply:
Sorry, on LOCAL can be used on NATIONAL no.

Reply:
We can use something like this
```typescript
toServicesTuple(new StrMap(serviceJson)).reduce(
    (acc, service: LocalServiceTuple | NationalServiceTuple) => {
      return {
        ...acc,
        [service.scope]: [...acc[service.scope], service]
      };
    },
```

Reply:
Sorry, maybe only now I've understood your original comment 😓

Reply:
I've removed the if with 341bb2b7b89d4e69f149c20f12e22f7a4433ca4a